core/services/mongodb-remote/src/test/java/com/mongodb/stitch/core/services/mongodb/remote/sync/internal/DataSynchronizerUnitTests.kt

```diff
@@ -121,4 +135,68 @@ class DataSynchronizerUnitTests {
 
         verify(remoteMongoCollection, times(1)).insertOne(any())
     }
+
+    @Test
+    fun testConfigure() {
+        // spy a new DataSynchronizer
+        val dataSynchronizer = spy(DataSynchronizer(
+            instanceKey,
+            service,
+            localClient,
+            remoteClient,
+            networkMonitor,
+            authMonitor
+        ))
+
+        // without a configuration it should not be
+        // configured or running
+        assertFalse(dataSynchronizer.isConfigured)
+        assertFalse(dataSynchronizer.isRunning)
+
+        // mock the necessary config args
+        val conflictHandler = mock(ConflictHandler::class.java) as ConflictHandler<BsonDocument>
+        val changeEventListener = mock(ChangeEventListener::class.java) as ChangeEventListener<BsonDocument>
+        val errorListener = mock(ErrorListener::class.java)
+        val bsonCodec = BsonDocumentCodec()
+
+        // fetch a new namespace, creating a new config
+        val nsConfig: NamespaceSynchronizationConfig = dataSynchronizer.getNamespaceConfig(namespace)
```

Comment:
[nit] I think you can get rid of `dataSynchronizer.getNamespaceConfig` and `getNamespaceConfig` as being exposed. Instead you should be able to utilize the top level public methods on the data synchronizer instead to act as a normal consumer which would make this a stronger test.

Reply:
Might it make more sense to do that in an integration test?

Reply:
I don't think so. The unit test here is aiming to check what happens if you are to configure twice. The first configuration should end up starting the listener. We don't need to reach that far into the implementation details. It ends up making the test brittle and not covering the actual aspect you want to test via the public API.